nixos/modules/services/networking/knot.nix

```diff
@@ -0,0 +1,94 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.knot;
+
+  configFile = pkgs.writeText "knot.conf" cfg.extraConfig;
+  socketFile = "/run/knot/knot.sock";
+
+  knotConfCheck = file: pkgs.runCommand "knot-config-checked"
+    { buildInputs = [ cfg.package ]; } ''
+    ln -s ${configFile} $out
+    knotc --config=${configFile} conf-check
+  '';
+in {
+  options = {
+    services.knot = {
+      enable = mkEnableOption "Knot authoritative-only DNS server";
+
+      extraArgs = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = ''
+          List of additional command line paramters for knotd
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = false;
```

Comment:
The default value of `false` causes an error when enabling the services without any configuration:

```
error: The option value `services.knot.extraConfig' in `/home/andi/dev/nixos/nixpkgs-review/nixos/modules/services/networking/knot.nix' is not of type `strings concatenated with "\n"'.
```

It should probably be "" or `types.nullOr types.lines` with some additional logic to handle the empty config file (if that is even valid).

Reply:
Knot seems to be fine with an empty config, so if that is sufficient to enable the service I'd stick with `""`.